### ISSUE TITLE
Inconsitencies with class_weight/sample_weight

### ISSUE THREAD

**Issue by nebw** [fchollet/keras#736]
In the documentation, the parameters class_weight/sample_weight are described as being applied only during training. Furthermore, the functions evaluate and test_on_batch don't have those parameters according to the documentation. 

Nevertheless, in the actual code, both function do have sample_weight but not class_weight as a parameter. This is unnecessarily confusing. Both functions should probably have either both or none of the parameters. Also, the documentation should be updated to reflect the actual interface.

I can implement those changes, but I'd have to know which solution is preferred. 

Also, in sklearn, class_weight can be set to 'auto'. This automatically calculates the weights based on the number of samples in each class. Would you accept a pull request for this feature?

**Comment by fchollet** [comment]
> I can implement those changes, but I'd have to know which solution is preferred.

Feel free to submit a PR for making the documentation consistent with the code.

> Both functions should probably have either both or none of the parameters.

Test functions cannot have a `class_weight` argument for the obvious reason that classes aren't known for test samples. But both can definitely have a `sample_weight` argument, and it is useful in a number of cases.

**Comment by nebw** [comment]
> Test functions cannot have a class_weight argument for the obvious reason that classes aren't known for test samples. But both can definitely have a sample_weight argument, and it is useful in a number of cases.

Can you explain what you mean by that? I must be missing something here. The classes are known for test samples, otherwise you couldn't compute the loss and there'd be no point in having the sample_weight argument either. It's also very trivial to add class_weight to the test functions (see https://github.com/nebw/keras/commit/42a34f9cf7cc3b8a3ef9b9b43b17428ad01bf945).

**Comment by fchollet** [comment]
I mean that weighting by classes in a test setup is a form of label information leak, because it cannot be reproduced in a real test situation (when you don't have the labels, such as when using `model.predict()`). It's acceptable during training, by definition, but should not be done at testing time.

**Comment by lemuriandezapada** [comment]
Well maybe all you have in terms of data is noisy, missing data. You might want to compute your error rate on just the points you have data for.

**Comment by nebw** [comment]
I see your point, even though I'm not sure I agree completely. class_weights essentially only modifies your loss function. There's no information leakage beyond that. This could be used to inadvertently overfit to the test data (e.g. by doing a grid search over class_weights and using early stopping based on the modified loss on the validation data) but the same could be said about sample_weight. In a real test situation, you probably wouldn't be able to use sample_weight either, and model.predict() consequently doesn't have that parameter.

I don't think there's a problem with using class_weights purely for testing (in contrast to validation during training), but I can see that adding this parameter might make it too easy to make such as mistake.

**Comment by JoaoLages** [comment]
What is the actual meaning of `sample_weight` while predicting? I though this parameter was related with how 'strong' we want to update the cost function for each sample
